email/notifier.go

```diff
@@ -74,13 +74,20 @@ func (a *AsyncWorkerNotifier) do(cn contextualNotification) {
 	receivers, vars, err := notification.Resolver(ctx, notification.ID)
 	if err != nil {
 		log.Error(ctx, map[string]interface{}{
-			"type": notification.ID,
-			"id":   notification.Type,
+			"type": notification.Type,
+			"id":   notification.ID,
 			"err":  err,
 		}, "failed to resolve receivers")
 
 		return
 	}
+	if receivers == nil || len(receivers) == 0 {
+		log.Info(ctx, map[string]interface{}{
+			"type": notification.Type,
+			"id":   notification.ID,
+		}, "skipping send email as no receivers resolved")
+		return
+	}
```

Comment:
if no_receiver then log info and return, no need to render email_body and call send_email call.